src/workloads/query/IDHack.yml

```diff
@@ -0,0 +1,235 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests performance of IDHACK.
+
+KeyWords:
+- IDHACK
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 400
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  MaxPhases: &MaxPhases 23
+
+ActorTemplates:
+- TemplateName: IDHACK
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "IDHackQuery"}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: 1}}
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: {unused: "please specify in which phases this actor should be active."}}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: 1000
+          Database: *Database
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: {_id: {$gt: 50, $lt: 100}}}}
+
+Actors:
+# Clear the collection before adding documents with a different type of _id.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0, 6, 12, 18]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: LoadIntIDs
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: 4800
+        BatchSize: 1000
+        Document:
+          _id: {^Inc: {start: 0, step: 1}}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2, 8, 14, 20]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- ActorFromTemplate:
+    TemplateName: IDHACK
+    TemplateParameters:
+      Name: IDHackOnIntID5Threads
+      Threads: 5
+      OnlyActiveInPhase: 3
+      Filter: {_id: {$gt: 50, $lt: 100}}
```

Comment:
As part of making this change, I think it would be good to use some sort of random number generator to spread the queries around, rather than repeatedly asking for the same document.

Reply:
Oh oops! I templated off the wrong one: https://github.com/mongodb/mongo-perf/blob/15526bd74a0ca44eba833c8d0ebb77db6a5c13dd/testcases/simple_query.js#L104

Okay I will change it to be a direct equality on random numbers!